Primo 2018.js

```diff
@@ -0,0 +1,428 @@
+					callArray.push(callNumber[i].textContent.match(/\$\$2\(?(.+?)(?:\s*\))?\$/)[1]);
+				}
+			}
+		}
+		if (callArray.length) {
+			//remove duplicate call numbers
+			callArray = dedupeArray(callArray);
+			item.callNumber = callArray.join(", ");
+		}
+		else {
+			ZU.xpathText(doc, '//enrichment/classificationlcc');
+		}
+	
+		if (pnxurl) {
+			item.libraryCatalog = pnxurl.match(/^https?:\/\/(.+?)\//)[1].replace(/\.hosted\.exlibrisgroup/, "");
+		}
+	
+		//Harvard specific code, requested by Harvard Library:
+		//Getting the library abbreviation properly,
+		//so it's easy to implement custom code for other libraries, either locally or globally should we want to.
+		var library;
+		var source = ZU.xpathText(doc, '//control/sourceid');
+		if (source) {
+			library = source.match(/^(.+?)_/);
+			if (library) library = library[1];
+		}
+		//Z.debug(library)
+		if (library && library == "HVD") {
+			if (ZU.xpathText(doc, '//display/lds01')) {
+				item.extra = "HOLLIS number: " + ZU.xpathText(doc, '//display/lds01');
+			}
+			if (ZU.xpathText(doc, '//display/lds03')) {
+				item.attachments.push({url: ZU.xpathText(doc, '//display/lds03'), title: "HOLLIS Permalink", snapshot: false});		
+			}
+		}
+		//End Harvard-specific code
+		item.complete();
+	});
+}
+
+
+
+
+function stripAuthor(str) {
+	return str
+		// Remove year
+		.replace(/\s*,?\s*\(?\d{4}-?(\d{4})?\)?/g, '')
+		// Remove things like (illustrator). TODO: use this to assign creator type?
+		.replace(/\s*,?\s*[\[\(][^()]*[\]\)]$/, '')
+		// The full "continuous" name uses no separators, which need be removed
+		// cf. "Luc, Jean André : de (1727-1817)"
+		.replace(/\s*:\s+/, " ");
+}
+
+function fetchCreators(item, creators, type, splitGuidance) {
+	for(var i=0; i<creators.length; i++) {
+		var creator = ZU.unescapeHTML(creators[i].textContent).split(/\s*;\s*/);
+		for(var j=0; j<creator.length; j++) {
+			var c = stripAuthor(creator[j]);
+			c = ZU.cleanAuthor(
+				splitGuidance[c.toLowerCase()] || c,
+				type,
+				true
+			);
+			
+			if (!c.firstName) {
+				delete c.firstName;
+				c.fieldMode = 1;
+			}
+			
+			item.creators.push(c);
+		}
+	}
+}
+
+function extractNumPages(str) {
+	// Borrowed from Library Catalog (PICA). See #756
+	//make sure things like 2 partition don't match, but 2 p at the end of the field do
+	// f., p., and S. are "pages" in various languages
+	// For multi-volume works, we expect formats like:
+	//   x-109 p., 510 p. and X, 106 S.; 123 S.
+	var numPagesRE = /\[?\b((?:[ivxlcdm\d]+[ ,\-]*)+)\]?\s+[fps]\b/ig,
+		numPages = [], m;
+	while(m = numPagesRE.exec(str)) {
+		numPages.push(m[1].trim()
+			.replace(/[ ,\-]+/g,'+')
+			.toLowerCase() // for Roman numerals
+		);
+	}
+	return numPages.join('; ');
+}
+
+function dedupeArray(names) {
+	//via http://stackoverflow.com/a/15868720/1483360
+	return names.reduce(function(a,b){
+		if(a.indexOf(b)<0) {
+			a.push(b);
+		}
+		return a;
+	},[]);
+}
+
+
+/** BEGIN TEST CASES **/
```

Comment:
We'll need to add more tests

Reply:
Hm... but the current tests don't work anymore. Hopefully we can have something stable here...